task-ballerina/commons.bal

```diff
@@ -14,118 +14,54 @@
-#                    of six or seven sub-expressions that describe individual details of the schedule. This
-#                    sub-expression is separated with white-space, and represent:
-#                       Seconds
-#                       Minutes
-#                       Hours
-#                       Day-of-Month
-#                       Month
-#                       Day-of-Week
-#                       Year (optional field)
-# + noOfRecurrences - Number of times to trigger the task after which the task stops running. If It is zero,
-#                     the task will be triggered forever.
-# + misfirePolicy - The policy, which is used to inform what it should do when a misfire occurs. The following are the
-#                   scenarios in which the policy can be used:
-#                       smartPolicy - This is the default policy, which will act as the `FireAndProceed`
-#                       ignorePolicy - If the Trigger misfires, this instructs the scheduler that the trigger will
-#                                      never be evaluated for a misfire situation and that the scheduler will
-#                                      simply try to fire it as soon as it can and then update the trigger as
-#                                      if it had fired at the proper time.
-#                       doNothing - Instructs the scheduler if the trigger misfires. Then, the trigger wants to have
-#                                   it's next-fire-time updated to the next time in the schedule after the current time.
-#                       fireAndProceed - Instructs the scheduler if the trigger misfires. Then, the trigger wants
-#                                        to be fired now by the scheduler.
-public type AppointmentConfiguration record {|
-    string cronExpression;
-    int noOfRecurrences = 0;
-    AppointmentMisfirePolicy misfirePolicy = "smartPolicy";
+public type JobId readonly & record {|
+   int id;
 |};
 
-# Possible types of parameters that can be passed into the `TimerTaskPolicy`.
-public type TimerMisfirePolicy RecurringTaskPolicy|OneTimeTaskPolicy;
+// Represents a ballerina job, which can be used to execute periodically
+public type Job object {
 
-# Possible types of parameters that can be passed into the `AppointmentTaskPolicy`.
-public type AppointmentMisfirePolicy SMART_POLICY|DO_NOTHING|FIRE_AND_PROCEED|IGNORE_POLICY;
+  // Calls by the Scheduler when a trigger fires
+  public function execute();
+};
+
+# Policies related to a trigger.
+#
+# + errorPolicy - The policy, which is used to inform what it should do when a error occurs.
+# + waitingPolicy - The policy, which is used to inform what it should do when waiting to get a resource
+#                   for the trigger.
+public type TaskPolicy record {|
+   ErrorPolicy? errorPolicy = LOG_AND_TERMINATE;
+   WaitingPolicy? waitingPolicy = IGNORE;
```

Comment:
If we are setting default values, why do we need to optional?
Let's make `WaitiingPolicy` default to `WAIT`